test_pii_data.py

```diff
@@ -42,63 +42,96 @@ def test_has_us_phone(self):
+        print("\n" + test_data.has_ipv4(True))
+
+        # Test address inside string
+        test_data = Pii('Samantha\'s address is 192.168.197.21')
+        self.assertTrue(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
 
         # Test a reserved address
         test_data = Pii('255.255.255.255')  # for broadcasting
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
         test_data = Pii('0.0.0.0')  # for default route
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
 
         # Test an out of range address
         test_data = Pii('192.168.168.256')
         self.assertFalse(test_data.has_ipv4())
+        holder = test_data.has_ipv4(True)
+        print("\n" + holder)
 
         # Test incorrect format
-        test_data = Pii('192.168.168.1.2.5')  # Test address with extra digits
-        self.assertFalse(test_data.has_ipv4())
         test_data = Pii('192.168')  # incomplete address
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
         test_data = Pii('192..168.168.256')  # extra dot
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
         test_data = Pii('.192.168.168.256')  # dot at beginning
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
         test_data = Pii('192.168.168.256.')  # dot at end
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
         test_data = Pii('1f2.168.168.256')  # with 'f' in place of number
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
         test_data = Pii('192.168.168.$')  # with '$' in place of number
         self.assertFalse(test_data.has_ipv4())
-        test_data = Pii('192,168,168,$')  # with incorrect delimiters(,)
+        print("\n" + test_data.has_ipv4(True))
+
+        test_data = Pii('192,168,168,2')  # with incorrect delimiters(,)
         self.assertFalse(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
         test_data = Pii('1.2.3')  # incomplete address
         self.assertFalse(test_data.has_ipv4())
-        # test_data = Pii('My IP address is 192.168.1.1')  # test an address embedded inside sentence
-        # self.assertTrue(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
+        test_data = Pii('My IP address is 192.168.1.1')  # test an address embedded inside sentence
+        self.assertTrue(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
+
+        test_data = Pii("Customer Sue Martin contacted the help desk to report "
+                        "an issue with their IPv4 address 192.168.185.88")  # test an address embedded inside sentence
+        self.assertTrue(test_data.has_ipv4())
+        print("\n" + test_data.has_ipv4(True))
```

Comment:
The print() statements are OK but they don't actually test anything.  Here is an example of how I would test one of the cases.  You need to implement something like this so you catch if anonymize is returning the correct value or not. 

```
        # Test anonymize
        self.assertEqual(test_data.has_ipv4(anonymize=True),
                         'Customer Sue Martin contacted the help desk to report an issue '
                         'with their IPv4 address [iPv4 address]')
```